src/commands/autocomplete/options.ts

```diff
@@ -129,7 +129,7 @@ export default class AutocompleteOptions extends AutocompleteBase {
   private get parsedFlagsWithEnvVars() {
     return Object.assign(
       {
-        app: process.env.HEROKU_APP,
+        app: process.env.HEROKU_APP || this.flags.app,
```

Comment:
@crcastle this fixes the app issue when inside a project, no need for `--app=` anymore to complete